### PR TITLE
Event listeners/resto druid

### PR DESCRIPTION
**Pull request by Zeboot**
Batch of changes for #3802 

once all batches are reviewed and merged, I'll disable legacy event listener support and enable the camel_case rule

Some formatting might be off due to large amount of changes. main things to look out for are inheritance and spell filters

### CODE REVIEW

File: src/parser/druid/restoration/constants.js [hunk 1]
```diff
@@ -15,19 +15,19 @@ export const ABILITIES_AFFECTED_BY_HEALING_INCREASES = [
   SPELLS.RENEWAL_TALENT.id,
   SPELLS.SPRING_BLOSSOMS.id,
   // The following spells don't double dip in healing increases.
-  SPELLS.LEECH.id,
+  SPELLS.LEECH,
```
Comment (MartijnHols) [inline src/parser/druid/restoration/constants.js]: This mismatches the lines above

File: src/parser/druid/restoration/modules/core/Rejuvenation.js [hunk 2]
```diff
@@ -18,24 +20,32 @@ class Rejuvenation extends Analyzer {
 
   totalRejuvsCast = 0;
 
-  on_byPlayer_heal(event) {
+  constructor(options){
+    super(options);
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.onHeal);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER), this.onCast);
```
Comment (MartijnHols) [inline src/parser/druid/restoration/modules/core/Rejuvenation.js]:
```suggestion
    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.REJUVENATION), this.onCast);
```

**Review by MartijnHols — CHANGES_REQUESTED** [review]

File: src/parser/druid/restoration/constants.js (hunk 1 above)
Comment (Zeboot) [reply src/parser/druid/restoration/constants.js]: I coulda sworn I undid those changes before I commited, let me fix that real quick

File: src/parser/druid/restoration/modules/features/Clearcasting.js [hunk 3]
```diff
@@ -33,37 +34,27 @@ class Clearcasting extends Analyzer {
   constructor(...args) {
     super(...args);
     this.procsPerCC = 1;
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.CLEARCASTING_BUFF), this.onApplyBuff);
+    this.addEventListener(Events.refreshbuff.by(SELECTED_PLAYER).spell(SPELLS.CLEARCASTING_BUFF), this.onRefreshBuff);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.CLEARCASTING_BUFF), this.onRemoveBuff);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.REGROWTH), this.onCast);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.REGROWTH), this.onHeal);
```
Comment (niseko) [inline src/parser/druid/restoration/modules/features/Clearcasting.js]:
```suggestion
    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.REGROWTH), this.onHeal);
```

**Review by niseko — CHANGES_REQUESTED** [review]

**Pull request merged**